.github/workflows/gh_pages.yml

```diff
@@ -0,0 +1,24 @@
+name: Deploy documentation to GH Pages
+
+on:
+  pull_request:
+  merge_group:
+  workflow_dispatch:
+  push:
+    branches: [main]
+
+jobs:
+  deploy-docs:
+    if: ${{ github.event_name == 'workflow_dispatch'  || github.event_name == 'merge_group'  }}
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: DeterminateSystems/nix-installer-action@main
+    - uses: DeterminateSystems/magic-nix-cache-action@main
+    - name: Build docs
+      run: nix build .#hax.docs --out-link ./built-docs
+    - name: Deploy to gh-pages
+      uses: peaceiris/actions-gh-pages@v3
```

Comment:
Let's deploy by action. Then we don't need to depend on third-party actions.
See for example https://github.com/cryspen/hacl-packages/blob/main/.github/workflows/gh-pages.yml#L142-L149
(this also needs to change of the page deployment setting to action)